src/libopensc/card-iasecc.c

```diff
@@ -867,8 +923,11 @@ iasecc_select_file(struct sc_card *card, const struct sc_path *path,
 		rv = iasecc_select_mf(card, file_out);
 		LOG_TEST_RET(ctx, rv, "MF selection error");
 
-		memmove(&lpath.value[0], &lpath.value[2], lpath.len - 2);
-		lpath.len -=  2;
+		if (!iasecc_is_cpx(card)) {
+			sc_log(ctx, "iasecc_select_mf done, remove 3f00 from the path");
+			memmove(&lpath.value[0], &lpath.value[2], lpath.len - 2);
+			lpath.len -=  2;
```

Comment:
If you *don't* remove 3F00 from the path, then you can remove the prior iasecc_select_mf()...

Reply:
I'll perform more tests with your advise and I'll provide some feedbacks.

Reply:
Thanks @frankmorgner  for the tips. Your are right, I can apply your advice and it works. Digging deeper on this topic, it allows to remove the patches 6a64f328499ec20e8d08f8dbcb1c77157b9f7065 and a60f4988f5fc108bae29945105b1e6e7b95b2331 from this serie too. I'll build an update of this pull request.